src/language_features/document_symbol.rs

```diff
@@ -23,6 +23,128 @@ use std::path::{Path, PathBuf};
 use unicode_width::UnicodeWidthStr;
 use url::Url;
 
+pub fn indent_guides(meta: EditorMeta, editor_params: EditorParams, ctx: &mut Context) {
```

Comment:
This doesn't work for blocks that are not indented.
For example this

```c++
namespace foo {
namespace bar {
namespace baz {

void foo() {
  int a;
  int b;
}

} // namespace baz
} // namespace bar
} // namespace foo
```

renders as

```
 namespace foo {
 namespace bar {
 │namespace baz {
 ┆│
 ┆│void foo() {
 ┊┆│  int a;
 ┊┆│  int b;
 ┆│}
 ┆│
 │} // namespace baz
 } // namespace bar
 } // namespace foo 
```

It's not critical to fix this if we don't enable it by default,
but I think it will be fine if we simply avoid adding an indent guide to the same column twice.

Reply:
Fixed by keeping lines and columns as keys in `BTree`s.